java-checks/src/main/java/org/sonar/java/checks/CompareStringsBoxedTypesWithEqualsCheck.java

```diff
@@ -19,29 +19,107 @@
  */
 package org.sonar.java.checks;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
 import org.sonar.check.Rule;
 import org.sonar.java.checks.helpers.ExpressionsHelper;
+import org.sonar.java.model.DefaultJavaFileScannerContext;
+import org.sonar.java.reporting.AnalyzerMessage;
+import org.sonar.java.reporting.InternalJavaIssueBuilder;
+import org.sonar.java.reporting.JavaQuickFix;
+import org.sonar.java.reporting.JavaTextEdit;
 import org.sonar.plugins.java.api.semantic.Type;
 import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
 import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
 
 @Rule(key = "S4973")
 public class CompareStringsBoxedTypesWithEqualsCheck extends CompareWithEqualsVisitor {
 
+  private static final String ISSUE_MESSAGE = "Strings and Boxed types should be compared using \"equals()\"";
+  private static final String QUICK_FIX_MESSAGE = "Replace with boxed comparison";
+  private static final String DOT_EQUALS_AND_OPENING_PARENTHESIS = ".equals(";
+
   @Override
   protected void checkEqualityExpression(BinaryExpressionTree tree) {
     Type leftOpType = tree.leftOperand().symbolType();
     Type rightOpType = tree.rightOperand().symbolType();
     if (!isNullComparison(leftOpType, rightOpType)
       && !isCompareWithBooleanConstant(tree.leftOperand(), tree.rightOperand())
       && (isStringType(leftOpType, rightOpType) || isBoxedType(leftOpType, rightOpType))) {
-      reportIssue(tree.operatorToken());
+      ((InternalJavaIssueBuilder) ((DefaultJavaFileScannerContext) this.context).newIssue())
+        .forRule(this)
+        .onTree(tree.operatorToken())
+        .withMessage(ISSUE_MESSAGE)
+        .withQuickFixes(() -> computeQuickFix(tree))
+        .report();
     }
   }
-  
+
   private static boolean isCompareWithBooleanConstant(ExpressionTree left, ExpressionTree right) {
     return ExpressionsHelper.getConstantValueAsBoolean(left).value() != null ||
       ExpressionsHelper.getConstantValueAsBoolean(right).value() != null;
   }
 
+  private static List<JavaQuickFix> computeQuickFix(BinaryExpressionTree tree) {
```

Comment:
No need to return a List when you have a single JavaQuickFix, you can directly return it, and call `withQuickFix`.